gstd/src/async_runtime/signals.rs

```diff
@@ -51,52 +51,107 @@ impl WakeSignals {
     }
 
     pub fn register_signal(&mut self, waiting_reply_to: MessageId) {
+        crate::log!("register_signal({waiting_reply_to:.2?})");
+
+        let message_id = crate::msg::id();
+
+        crate::log!(
+            "register_signal({waiting_reply_to:.2?}): inserting signal for {message_id:.2?}"
+        );
+
         self.signals.insert(
             waiting_reply_to,
             WakeSignal {
-                message_id: crate::msg::id(),
+                message_id,
                 payload: None,
                 waker: None,
             },
         );
     }
 
     pub fn record_reply(&mut self) {
-        if let Some(signal) = self
-            .signals
-            .get_mut(&crate::msg::reply_to().expect("Shouldn't be called with incorrect context"))
-        {
-            signal.payload = Some((
-                crate::msg::load_bytes().expect("Failed to load bytes"),
-                crate::msg::reply_code().expect("Shouldn't be called with incorrect context"),
-            ));
-
-            if let Some(waker) = &signal.waker {
-                waker.wake_by_ref();
-            }
+        let reply_to = crate::msg::reply_to().expect("Shouldn't be called with incorrect context");
+
+        crate::log!("record_reply({reply_to:.2?})");
+
+        let Some(signal) = self.signals.get_mut(&reply_to) else {
+            crate::log!("record_reply({reply_to:.2?}): signal is NOT FOUND");
+            return;
+        };
+
+        crate::log!("record_reply({reply_to:.2?}): signal is FOUND");
 
-            crate::exec::wake(signal.message_id).expect("Failed to wake the message")
+        crate::log!("record_reply({reply_to:.2?}): querying data");
+
+        let payload = crate::msg::load_bytes().expect("Failed to load bytes");
+        let reply_code =
+            crate::msg::reply_code().expect("Shouldn't be called with incorrect context");
+
+        crate::log!("record_reply({reply_to:.2?}): setting data to signal");
+
+        signal.payload = Some((payload, reply_code));
+
+        crate::log!("record_reply({reply_to:.2?}): trying to touch waker");
```

Comment:
I think it is yet another redundant message which can be easily combined with the above from line 90